app/components/search/Sidebar/Sidebar.tsx

```diff
@@ -36,6 +36,29 @@ const Sidebar = ({
   setIsMapCollapsed: (_isMapCollapsed: boolean) => void;
 }) => {
   const [filterMenuVisible, setfilterMenuVisible] = useState(false);
+  const filterMenuRef = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (
+      filterMenuRef.current &&
+      !filterMenuRef.current.contains(event.target as Node)
+    ) {
+      setfilterMenuVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    if (filterMenuVisible) {
+      document.addEventListener("mousedown", handleClickOutside);
+    } else {
+      document.removeEventListener("mousedown", handleClickOutside);
+    }
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [filterMenuVisible]);
+
```

Comment:
What about creating a hook called `useClickOutside` or something for mixing in this logic? Do you think it's something we would reuse elsewhere? It feels like the kind of implementation detail is nice to hide behind an abstraction.